File: back-end/src/main/java/com/example/backend/models/Matiere.java
package com.example.backend.models;

import com.example.backend.enumeration.*;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Matiere {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nomMatiere;

    @Enumerated(EnumType.STRING)
    private DepartementEnum departement;

    @Enumerated(EnumType.STRING)
    private SpecialiteEnum specialite;

    private Boolean recu = false;
    private Boolean valide = false;

    @Enumerated(EnumType.STRING)
    private NiveauEnum niveau;

    @Enumerated(EnumType.STRING)
    private GroupeEnum groupe;

    @Enumerated(EnumType.STRING)
    private SemesterEnum semester;

    @Enumerated(EnumType.STRING)
    private TypeMatiereEnum TypeMatiere;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "enseignant_id")
    private Enseignant enseignant;
}

File: front-end/src/app/models/matiere.ts
export interface MatiereBackend {
  id: number;
  nomMatiere: string;
  departement: string;
  specialite: string;
  niveau: string;
  groupe: string;
  semester: string;
  typeMatiere: string;
  enseignantId: number;
  enseignantNom: string;
  enseignantEmail: string;
  recu?: boolean;
  valide?: boolean;
}

export interface EnseignantInMatiere {
  id: number;
  nom: string;
  email: string;
}

export interface MatiereFrontend {
  id: number;
  nomMatiere: string;
  departement: string;
  specialite: string;
  niveau: string;
  groupe: string;
  semester: string;
  typeMatiere: string;
  enseignant: EnseignantInMatiere;
  recu: boolean;
  valide: boolean;
}


